tools/shared.py

```diff
@@ -1259,8 +1259,8 @@ def verify_settings():
       # TODO(sbc): Make this into a hard error.  We still have a few places that
       # pass WASM=0 before we can do this (at least Platform/Emscripten.cmake and
       # generate_struct_info).
-      logger.warn('emcc: WASM_BACKEND is not compatible with asmjs (WASM=0), forcing WASM=1')
       Settings.WASM = 1
+      Settings.WASM2JS = 1
```

Comment:
Not sure how I feel about `WASM=0` meaning `WASM2JS=1`, and also having `WASM2JS` as a setting. This leaves us with two ways to set WASM2JS, which lets us land in weird places.
Say the user says `emcc foo.c -s WASM=0 -s WASM2JS=0`. There's not actually a reasonable thing to do here (maybe error out?), and what this code does do sets WASM2JS to 1, which may or may not be what the user intended.

I'm ok with either WASM=0 meaning "give me JS" and wasm2js is an implementation detail, or `-s WASM2JS=1` and we continue to ignore (and warn about) WASM=0, but I find two flags problematic. Something something DRY principle

Reply:
Yeah, not sure what's best for the flags. I wonder if maybe `WASM=0` is useful for backwards compatibility, and justifies a warning, while `WASM2JS=1` explicitly means the user understands and wants this new code path, so we don't need the warning?